search/api.py

```diff
@@ -1,306 +1,142 @@
-        'certificates': {'type': 'nested'}
-    })
-    # Make strings not_analyzed by default
-    mapping.meta('dynamic_templates', [{
-        "notanalyzed": {
-            "match": "*",
-            "match_mapping_type": "string",
-            "mapping": NOT_ANALYZED_STRING_TYPE
-        }
-    }])
-    return mapping
+    def add_to_results(self, es_search_hits):
+        """
+        Adds a set of search results ('hits') to the combined results
+        """
+        self.results = self.results + list(es_search_hits)
 
+    def execute(self, search_obj, page_size=DEFAULT_ES_LOOP_PAGE_SIZE):
+        """
+        Executes a search query, pages through all of its results, maintains
+        a record of all results found, and returns those combined results.
+        """
+        self.init_combined_results()
+        # Maintaining a consistent sort on '_doc' will help prevent bugs where the
+        # index is altered during the loop
+        search_obj.sort('_doc')
+        loop = 0
+        all_results_returned = False
+        while not all_results_returned:
+            from_index = loop * page_size
+            to_index = from_index + page_size
+            search_results = execute_search(search_obj[from_index: to_index])
+            self.add_to_results(search_results)
+            all_results_returned = to_index >= search_results.hits.total
+            loop += 1
+        return self.results
 
-def create_program_enrolled_user_mapping():
-    """
-    Save the mapping for a program user. If one already exists, delete it first.
-    """
-    conn = get_conn(verify=False)
-    index_name = settings.ELASTICSEARCH_INDEX
-    if conn.indices.exists_type(index=index_name, doc_type=USER_DOC_TYPE):
-        conn.indices.delete_mapping(index=index_name, doc_type=USER_DOC_TYPE)
-    mapping = program_enrolled_user_mapping()
-    mapping.save(index_name)
+    # Make the objects callable - ie: invoking an object of this class directly
+    # will call the given method.
+    __call__ = execute
 
 
-def create_mappings():
+class ESFullResultEmailSearch(ESFullResultSearch):
     """
-    Create all mappings, deleting existing mappings if they exist.
+    Provides a function to return all unique emails for documents that match
+    an ES query
     """
-    create_program_enrolled_user_mapping()
+    def init_combined_results(self):
+        self.results = set()
 
+    def add_to_results(self, es_search_hits):
+        for hit in es_search_hits:
+            self.results.add(hit.email)
 
-def clear_index():
-    """
-    Wipe and recreate index and mapping. No indexing is done.
-    """
-    conn = get_conn(verify=False)
-    index_name = settings.ELASTICSEARCH_INDEX
-    if conn.indices.exists(index_name):
-        conn.indices.delete(index_name)
-    conn.indices.create(index_name)
-    conn.indices.refresh()
-    create_mappings()
+    def execute(self, search_obj, page_size=DEFAULT_ES_LOOP_PAGE_SIZE):
+        search_obj = search_obj.fields('email')
```

Comment:
The code coverage doesn't show these two lines being executed during the tests... I think your `__call__` assignment is not inherited here